src/main/java/exercise39/SortingRecords.java

```diff
@@ -45,27 +47,20 @@ public String recordSorter(List<Map<String, String>> records) {
         records.sort(Comparator.comparing(record -> record.get("Last Name")));
 
         for (Map<String, String> map : records) {
-//            String date = map.get("Date");
             unsortedRecords += map.get("First Name") + " " + map.get("Last Name") + " " + map.get("Position") + "" + map.get("Date");
-
-//            if (date != null && !date.isEmpty()) {
-//                unsortedRecords += " " + date;
-//        }
-        unsortedRecords += "\n";
-    }
+            unsortedRecords += "\n";
+        }
         return unsortedRecords.trim();
     }
 
     public String recordFilter(String searchTerm, List<Map<String, String>> records) {
 
         String filteredRecords = "";
-//        for (Map<String, String> map : records) {
-//            if (map.get("First Name").contains(searchTerm) || map.get("Last Name").contains(searchTerm)) {
-//                filteredRecords += map.get("First Name") + " " + map.get("Last Name") + " " + map.get("Position");
-//            }
+
         records.removeIf(map -> !map.get("First Name").contains(searchTerm) && !map.get("Last Name").contains(searchTerm));
 
         for (Map<String, String> map : records) {
+            //repetition still - and no date?
```

Comment:
So we're still duplicating building up the String that represents each record, here and a few lines above - can we remove the repetition?